src/fetch/transfer_file.py

```diff
@@ -402,13 +403,17 @@ def get_cow_rewards(dune: DuneAPI, start_block: str, end_block: str) -> list[Tra
     solver_num_trades: dict[str, int] = {
         row["solver"].lower(): int(row["num_trades"]) for row in dune_trade_counts
     }
-    assert set(cow_rewards_df.receiver) == set(
-        solver_num_trades.keys()
-    ), "solver set != receiver set"
-    for row in cow_rewards_df.rows:
+    db_solvers = set(cow_rewards_df.receiver)
+    dune_solvers = set(solver_num_trades.keys())
+    assert (
+        db_solvers == dune_solvers
+    ), f"solver sets disagree: {db_solvers.symmetric_difference(dune_solvers)}"
+    for _, row in cow_rewards_df.iterrows():
         solver = row.receiver.lower()
         assert row.num_trades == solver_num_trades[solver], "invalid trade count!"
```

Comment:
Yes I like this. Unfortunately I didn't see you review till it was merged. I will fix all these things in a follow up.

Reply:
> Yes I like this. Unfortunately I didn't see you review till it was merged. I will fix all these things in a follow up.

Yeah I saw that very late too. I just took longer than usual because of the size of the PR. Good to know you will do that. Thanks!